### ISSUE TITLE
Allow rest parameter in mixin

### ISSUE THREAD

**Issue by curvedmark** [curvedmark/roole#10]
```
$mixin = @mixin $param, ...$params
    body
        margin: 0
```

**Comment by jamiebuilds** [comment]
The ellipsis should probably come afterwards for readability

**Comment by jamiebuilds** [comment]
Could there also be a way of targeting the params inside the mixin like `params[1]`, `params:2` (my preference), or `params-3`

**Comment by curvedmark** [comment]
I want to follow [javascript's syntax](https://developer.mozilla.org/en-US/docs/JavaScript/Reference/rest_parameters).

Is the `$arguments` variable (see [the language spec](http://roole.org/#mixin)) what you are looking for? It will be implemented in v0.3.0

**Comment by jamiebuilds** [comment]
Well Sass, stylus, and I believe Less have `$arguments` and `$args...`, and they all have various ways of getting a particular argument (like sass has `nth($args, 2)`). I'd like to have the javascript style `$args[1]` of getting a particular argument.

So `$args...` or `$arguments` would give you an array/list. And Roole would allow you to target an array item with a `[#]`

**Comment by curvedmark** [comment]
good idea, and negative index should also be supported.

**Comment by curvedmark** [comment]
Landed in e86b79f8cd72b6d0ce0660eef0359b7559af9b1a
